### PR TITLE
Add support to api_level based OS releases and read parameter from elf metadata

### PR DESCRIPTION
**Pull request by xchapron-ledger**
This PR does:
- Add support to `apiLevel` parameter needed for next firmwares. It is replacing the `sdk` parameter.
- Add autodetection of `model` and `sdk` / `apiLevel` parameter based on elf metadata.
- Change default `model` value from `nanos` to `None`
- Add support to `apiLevel` 1 firmware on `nanosp` and `nanox`

### CODE REVIEW

File: speculos/main.py [hunk 1]
```diff
@@ -46,6 +46,28 @@ def set_pdeath(sig):
...
+    with open(app_path, 'rb') as fp:
+        elf = ELFFile(fp)
+        metadata = {}
+        for section_name in ELF_METADATA_SECTIONS:
+            section = elf.get_section_by_name(f"ledger.{section_name}")
+            if section:
+                metadata[section_name] = section.data().decode("utf-8").strip()
+                if section_name == "target":
+                    if metadata[section_name] == "nanos2":
+                        metadata[section_name] = "nanosp"
```
Comment (cbouvet-ledger) [inline speculos/main.py:67]: why not just change the nanosp to nanos2 in speculos?

File: speculos/main.py [hunk 2]
```diff
@@ -93,17 +115,25 @@ def run_qemu(s1: socket.socket, s2: socket.socket, args: argparse.Namespace) ->
...
+    else:
+        argv += ['-k', str(args.sdk)]
 
-    # load cxlib only if available for the specified sdk
-    cxlib = pkg_resources.resource_filename(__name__, f"/cxlib/{args.model}-cx-{args.sdk}.elf")
+    # load cxlib only if available for the specified api level or sdk
+    if args.apiLevel:
+        cxlib_filepath = f"/cxlib/{args.model}-api-level-cx-{args.apiLevel}.elf"
+    else:
+        cxlib_filepath = f"/cxlib/{args.model}-cx-{args.sdk}.elf"
```
Comment (cbouvet-ledger) [inline speculos/main.py:129]: sdk should be deprecated in a later PR

File: speculos/main.py [hunk 3]
```diff
@@ -220,7 +251,37 @@ def main(prog=None):
     if prog:
         parser.prog = prog
     args = parser.parse_args()
-    args.model.lower()
+
+    if args.model:
+        args.model.lower()
```
Comment (cbouvet-ledger) [inline speculos/main.py:256]: args.model = args.model.lower()

**Review by cbouvet-ledger — APPROVED** [review]

File: speculos/main.py (hunk 1 above)
Comment (xchapron-ledger) [reply speculos/main.py:67]: Not a fight for today :)

**Pull request merged**